drivers/gpio/gpio_stm32.c

```diff
@@ -636,7 +636,7 @@ static int gpio_stm32_init(const struct device *dev)
 			    &gpio_stm32_data_## __suffix,		       \
 			    &gpio_stm32_cfg_## __suffix,		       \
 			    PRE_KERNEL_1,				       \
-			    CONFIG_KERNEL_INIT_PRIORITY_DEFAULT,	       \
+			    CONFIG_GPIO_INIT_PRIORITY,			       \
```

Comment:
This change is moving prio from 40 (KERNEL_INIT_PRIORITY_DEFAULT) to 50 (KERNEL_INIT_PRIORITY_DEVICE).
This is a concern since GPIO init should take place before some other devices like UART in specific cases (PM=y).
So at least for stm32, `GPIO_INIT_PRIORITY` should be redefined to `KERNEL_INIT_PRIORITY_DEFAULT` either in `drivers/gpio/Kconfig.stm32` or `soc/arm/st_stm32/common/Kconfig.defconfig.series`.

And, since you're applying the same change to other drivers (stellaris, rv32m1, ...), I guess the same should be applied for those as well.